src/FakeItEasy/Analysis/MustUseReturnValueAttribute.cs

```diff
@@ -0,0 +1,25 @@
+﻿namespace FakeItEasy.Analysis
+{
+    using System;
+
+    /// <summary>
+    /// Indicates that the return value of a method should not be ignored.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Method, AllowMultiple = false, Inherited = true)]
+    public sealed class MustUseReturnValueAttribute : Attribute

```

Comment:
I would either try the friend approach, or failing that would not object to accessing the attribute by reflection. I did it in my spike, and it seemed to work.